x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -19,14 +21,19 @@ func (k Keeper) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake) (*
 	if err != nil {
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "address is invalid: %s", msg.Creator)
 	}
-
-	// TODO(TEST-112) add safety check to validate the receiver address is a valid hostZone address
 	// then make sure host zone is valid
 	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
 	if !found {
 		return nil, sdkerrors.Wrapf(types.ErrInvalidHostZone, "host zone is invalid: %s", msg.HostZone)
 	}
 
+	// ensure the recipient address is a valid bech32 address on the hostZone
+	// TODO(TEST-112) do we need to check the hostZone before this check? Would need access to keeper 
+	_, err = utils.AccAddressFromBech32(msg.Receiver, hostZone.Bech32Prefix)
```

Comment:
Should this live in `ValidateBasic`? I don't have a great sense of where to put these validity checks. On the one hand, it's logically cleaner to put it in `ValidateBasic`, but on the other we'd have to read in the host zone twice

Reply:
Never mind, per the docs `ValidateBasic can include only stateless checks (the checks that do not require access to the state`